src/main/scala/it/pagopa/interop/agreementprocess/api/impl/AgreementApiServiceImpl.scala

```diff
@@ -1,418 +1,517 @@
+      handleCreationError(
+        s"Error creating agreement for EService ${payload.eserviceId} and Descriptor ${payload.descriptorId}"
+      ) orElse { case Success(agreement) => createAgreement200(agreement) }
     }
   }
 
-  /** Code: 204, Message: Active agreement suspended.
-    * Code: 400, Message: Bad Request, DataType: Problem
-    */
-  override def suspendAgreement(agreementId: String, partyId: String)(implicit
+  override def submitAgreement(agreementId: String)(implicit
     contexts: Seq[(String, String)],
-    toEntityMarshallerProblem: ToEntityMarshaller[Problem]
-  ): Route = authorize(ADMIN_ROLE) {
-    logger.info("Suspending agreement {}", agreementId)
-    val result = for {
-      agreement          <- agreementManagementService.getAgreementById(agreementId)
-      changeStateDetails <- AgreementManagementService.getStateChangeDetails(agreement, partyId)
-      _                  <- agreementManagementService.suspendById(agreementId, changeStateDetails)
-      _                  <- authorizationManagementService.updateStateOnClients(
-        eServiceId = agreement.eserviceId,
-        consumerId = agreement.consumerId,
-        agreementId = agreement.id,
-        state = AuthorizationManagementDependency.ClientComponentState.INACTIVE
-      )
-    } yield ()
+    toEntityMarshallerProblem: ToEntityMarshaller[Problem],
+    toEntityMarshallerAgreement: ToEntityMarshaller[Agreement]
+  ): Route =
+    authorize(ADMIN_ROLE) {
+      logger.info("Submitting agreement {}", agreementId)
+      val result = for {
+        requesterOrgId <- getRequesterOrganizationId(contexts)
+        agreement      <- agreementManagementService.getAgreementById(agreementId)
+        _              <- assertRequesterIsConsumer(requesterOrgId, agreement)
+        _              <- agreement.assertSubmittableState.toFuture
+        _              <- verifySubmissionConflictingAgreements(agreement)
+        eService       <- catalogManagementService.getEServiceById(agreement.eserviceId)
+        _              <- CatalogManagementService.validateSubmitOnDescriptor(eService, agreement.descriptorId)
+        consumer       <- tenantManagementService.getTenant(agreement.consumerId)
+        _              <- validateDeclaredAttributes(eService, consumer)
+        updated        <- submit(agreement, eService, consumer)
+        _              <- validateCertifiedAttributes(eService, consumer) // Just to return the error. TODO required?
+      } yield updated.toApi
 
-    onComplete(result) {
-      case Success(_)  => suspendAgreement204
-      case Failure(ex) =>
-        logger.error(s"Error while suspending agreement $agreementId", ex)
-        val errorResponse: Problem =
-          problemOf(StatusCodes.BadRequest, SuspendAgreementError(agreementId))
-        suspendAgreement400(errorResponse)
+      onComplete(result) {
+        handleSubmissionError(s"Error while submitting agreement $agreementId") orElse { case Success(agreement) =>
```

Comment:
"Yet another error handling?!"
I know... I know... 
You are right, but until we decide a way, I'll make experiments 😋